src/docfx/publish/PublishUrlMap.cs

```diff
@@ -132,16 +132,34 @@ private PublishUrlMapItem ResolveOutputPathConflicts(IGrouping<string, PublishUr
             }
 
             // redirection file is preferred than source file
-            var redirection = conflicts.FirstOrDefault(x => x.SourcePath.Origin == FileOrigin.Redirection);
-            if (redirection != null)
+            var redirections = conflicts.Where(x => x.SourcePath.Origin == FileOrigin.Redirection).OrderBy(x => x.SourcePath.Path, PathUtility.PathComparer);
+            var nonRedirections = conflicts.Where(x => x.SourcePath.Origin != FileOrigin.Redirection)
+                .OrderBy(x => x.SourcePath.Path, PathUtility.PathComparer)
+                .Select(x => x.SourcePath.Path.Value).ToList();
+            var redirection = redirections.FirstOrDefault();
```

Comment:
nit: consider put this declaration into a block where `redirectionCount > 0`, then we can get rid of all the nullable handling

Reply:
There are 2 parts that satisfy `redirectionCount > 0`. I don't want too many nested `if...else...`